### PR TITLE
feat: template field filtering

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Activates template field filtering, which was already in the code but not applied.

## Context:

This is applied to all handlebars templates.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/regex/index.ts [hunk 1]
```diff
@@ -47,7 +47,12 @@ function createDependency(
     const fieldTemplate = `${field}Template`;
     if (config[fieldTemplate]) {
       try {
-        dependency[field] = template.compile(config[fieldTemplate], groups);
+        const filterFields = false;
```
Comment (viceice) [inline lib/manager/regex/index.ts:50]: Why extra variable?

Comment (rarkins) [reply lib/manager/regex/index.ts:50]: The input to this compilation is the matching regex capture groups, which (a) is their own data, and (b) can be named anything they want anyway, so filtering on name seems pointless

Comment (viceice) [reply lib/manager/regex/index.ts:50]: so way hon simply pass `false` to compile funtion?

Comment (rarkins) [reply lib/manager/regex/index.ts:50]: ?

File: lib/manager/regex/index.ts [hunk 2]
```diff
@@ -47,7 +47,12 @@ function createDependency(
...
+        dependency[field] = template.compile(
+          config[fieldTemplate],
+          groups,
+          filterFields
+        );
```
Comment (viceice) [inline lib/manager/regex/index.ts:55]:
```suggestion
        dependency[field] = template.compile(
          config[fieldTemplate],
          groups,
          false
        );
```
Isn't this enough? Or do you need `filterFields` const later?

Comment (rarkins) [reply lib/manager/regex/index.ts:55]: I thought it was clearer this way

**Review by viceice — APPROVED** [review]

**Pull request merged**